### ISSUE TITLE
Measure code coverage for Gradle plugin

### ISSUE THREAD

**Issue by 3flex** [detekt/detekt#1294]
## Expected Behavior
Test coverage for the Gradle plugin is measured by JaCoCo. Results are uploaded to Codecov after a successful CI build.

## Current Behavior
Neither happens.

## Context
Tracking code coverage for all modules means coverage and hopefully quality will improve over time.

Unfortunately JaCoCo doesn't run out of the box on tests that use Gradle TestKit, so this will require a bit of trickery to get working properly.

**Comment by 3flex** [comment]
Blocked until https://github.com/arturbosch/detekt/issues/1754 is resolved.

**Comment by schalkms** [comment]
Codecov is fine again. Therefore, I'm closing this.

**Comment by 3flex** [comment]
Not quite, the unit tests (those not relying on Gradle TestKit) result in coverage, but not the integration tests which don't generate coverage info due to https://github.com/gradle/gradle/issues/1465

So yes, there's some coverage info, but it's not complete. You can see for example coverage info for the Detekt task which shows no coverage of the `@TaskAction`: https://codecov.io/gh/detekt/detekt/src/e3fd6c88cc0477084de3e69de7c901390683ac08/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/Detekt.kt#L203

**Comment by 3flex** [comment]
It would still be good to do this

**Comment by BraisGabin** [comment]
Is this even doable from our side? Shouldn't we just wait for https://github.com/gradle/gradle/issues/1465 or just accept that we are not going to get this reports?

**Comment by 3flex** [comment]
There's a plugin we could try. Let's keep this open.
